harness/determined/cli/trial.py

```diff
@@ -125,71 +125,72 @@ def trial_logs(args: Namespace) -> None:
-
-    trial_logs_file = write_trial_logs(args, args.trial_id, output_dir_tar)
-    output_dir_tar.add(f'{trial_logs_file}.json')
-    os.remove(f'{trial_logs_file}.json')
-    
-    master_logs_file = write_master_logs(args, output_dir_tar)
-    output_dir_tar.add(f'{master_logs_file}.json')
-    os.remove(f'{master_logs_file}.json')
-
-
-    api_trail_file, api_experiment_file = write_api_call(args, output_dir_tar)
-    output_dir_tar.add(f'{api_trail_file}.json')
-    output_dir_tar.add(f'{api_experiment_file}.json')
-    os.remove(f'{api_trail_file}.json')
-    os.remove(f'{api_experiment_file}.json')
-
-    output_dir_tar.close()
-
-    return f'{args.output_dir}.tar.gz'
-
-    
-
-def write_trial_logs(args,trial_id, output_dir):  #difference between this trial_logs and api call? There doesn't seem to be any. 
-    trial_logs = api.trial_logs(args.master, trial_id)
-    # use trial logs to get the experiment id 
-    file_name = 'trial_logs'
-    trial_logs_list = []
-    for log in trial_logs: 
-        trial_logs_list.append(log)
-
-    create_json_file_in_dir(trial_logs_list,file_name, output_dir)
-    return file_name
-
-def write_master_logs(args, output_dir):
-    response = api.get(
-                args.master, "logs"
-            )
-    file_name = 'master_logs'
-    create_json_file_in_dir(response.json(),file_name, output_dir)
-    return file_name
-
-def write_api_call(args, output_dir): 
-    file_name1 = 'api_experiment_call'
-    file_name2 = 'api_trial_call'
+def output_logs(args: Namespace):
+    temp_dir = tempfile.TemporaryDirectory()
+    trial_logs_filepath = write_trial_logs(args, temp_dir)
+
+    master_logs_filepath = write_master_logs(args, temp_dir)
+
+    api_experiment_filepath, api_trail_filepath = write_api_call(args, temp_dir)
+
+    bundle = None
+    if args.output_dir is None:
+        output_dir = os.getcwd()
+        tar = f"det-bundle-trail_{args.trial_id}-{datetime.now().strftime('%Y-%m-%d_%H:%M:%S')}.tar.gz"
+        fullpath = os.path.join(output_dir, tar)
+        bundle = tarfile.open(fullpath, "w:gz")
+
+    else:
+        bundle = tarfile.open(f"{args.output_dir}.tar.gz", "w:gz")
+
+    bundle.add(trial_logs_filepath, arcname=os.path.basename(trial_logs_filepath))
+    bundle.add(master_logs_filepath, arcname=os.path.basename(master_logs_filepath))
+    bundle.add(api_trail_filepath, arcname=os.path.basename(api_trail_filepath))
+    bundle.add(api_experiment_filepath, arcname=os.path.basename(api_experiment_filepath))
+
+    bundle.close()
+
+
+def write_trial_logs(args, temp_dir):
+    trial_logs = api.trial_logs(args.master, args.trial_id)
+    file_path = os.path.join(temp_dir.name, "trial_logs.jsonl")
+    create_jsonl_file_in_dir(trial_logs, file_path)
+    return file_path
+
+
+def write_master_logs(args, temp_dir):
+    response = api.get(args.master, "logs")
+    file_path = os.path.join(temp_dir.name, "master_logs.jsonl")
+    create_json_file_in_dir(response.json(), file_path)
```

Comment:
I looked into this I'm not calling the logs function in harness/determined/cli/master.py. I'm only making the api call: api.get(args.master, "logs") and this will trigger the function getMasterLogs in master/internal/core.go and I don't see any sleeps over there so I don't think we need to handle polling.